### PR TITLE
WIP Migrate Fat Free CRM to Rails 7

### PR DESCRIPTION
**Pull request by johnbumgardner**
6.1->7.0

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -86,8 +86,8 @@ group :heroku do
   gem 'puma'
 end
 
-gem 'responds_to_parent', git: 'https://github.com/CloCkWeRX/responds_to_parent.git', branch: 'patch-2' # Temporarily pointed at git until https://github.com/zendesk/responds_to_parent/pull/7 is released
-gem 'acts_as_commentable', git: 'https://github.com/fatfreecrm/acts_as_commentable.git', branch: 'rails-61' # Our fork
+gem 'responds_to_parent'
+gem 'acts_as_commentable'
```
Comment (CloCkWeRX) [inline Gemfile:90]: Be aware that the upstream maintainer of this gem went MIA.

https://github.com/jackdempsey/acts_as_commentable/compare/master...fatfreecrm:acts_as_commentable:rails-61

I'd recommend forking from our version and making a rails-7 branch

File: Gemfile [hunk 2]
```diff
@@ -86,8 +86,8 @@ group :heroku do
   gem 'puma'
 end
 
-gem 'responds_to_parent', git: 'https://github.com/CloCkWeRX/responds_to_parent.git', branch: 'patch-2' # Temporarily pointed at git until https://github.com/zendesk/responds_to_parent/pull/7 is released
-gem 'acts_as_commentable', git: 'https://github.com/fatfreecrm/acts_as_commentable.git', branch: 'rails-61' # Our fork
+gem 'responds_to_parent'
```
Comment (CloCkWeRX) [inline Gemfile:89]: Ditto this one. Zendesk appear to be a bit useless at merging.

https://github.com/zendesk/responds_to_parent/pull/7

File: app/models/fields/custom_field.rb [hunk 3]
```diff
@@ -49,6 +49,8 @@
 #   since changing the type of the database column would cause data to be lost.
 #
 
+require_relative 'field'
```
Comment (CloCkWeRX) [inline app/models/fields/custom_field.rb:52]: I assume this is something to do with the zeitgiest autoloader changes?

May be worth doing bundle exec zeitgiest:check and submitting all that as a standalone PR; then doing Rails 7 atop that.

**Review by CloCkWeRX — CHANGES_REQUESTED** [review]
Might need to be careful not to accidentally nuke the various copyright notices.

File: app/models/fields/custom_field.rb (hunk 3 above)
Comment (johnbumgardner) [reply app/models/fields/custom_field.rb:52]: yeah im not sure why exactly. still working through this. ActiveRecord::Base.descendents end up having far less classes accounted for at any given instance

**Comment by hubyhuby** [comment]
Hope I am not disturbing here the google group is inactive for 1 year. 
I am starting a green field project witgh fat free, I have started a onclick install for ruby on rails.
But digital Ocean now only offers rails 7.0. and fat free requires for now 6.1

I am wonderning if anyone has experience with Digital Ocean ?
Will it make any troubles running fat free on this instance ? 
Thks for your help

https://www.digitalocean.com/community/tutorials/how-to-use-the-ruby-on-rails-1-click-install-on-digitalocean
image

Thks for this great CRM !

**Comment by ferrisoxide** [comment]
@hubyhuby 

> I am wonderning if anyone has experience with Digital Ocean ?
> Will it make any troubles running fat free on this instance ?
> Thks for your help

I don't think it will work out of the box, simply because the default version of Ruby used by FatFreeCRM (Ruby 2.7.5) probably won't be supported on DO. I could be wrong - I haven't used the platform for a while - but as 2.7.5 has now reached EOL it's quite likely it won't be installable.

Having said that, the OneClick Install might bring up enough of the dependencies required for running FFCrm. You would have to muck around with the source (_e.g._ change `.ruby-version` to use 3.2.0) to get the code to run - but TBH I think our best bet is to bring the source up to spec with more recent versions of Ruby. I've created a PR to bring it up to 3.0.2 (see #1139) but it should be trivial to migrate to 3.2.0.

As an aside, you can install whatever you want on DO droplets - within reason. If you want an older Ruby you can install it by hand - assuming the distro you use still supports it - or by using a provisioning tool like Ansible, Chef, etc.

**Comment by CloCkWeRX** [comment]
Probably broke things, but well, 3.1 is in so this is rebased to that.

**Comment by steveyken** [comment]
Can this PR close now that FFCRM is Rails 7 or is there some outstanding code here to merge?

**Pull request closed without merging**